numpyro/distributions/constraints.py

```diff
@@ -638,6 +639,15 @@ def feasible_like(self, prototype):
         return jax.numpy.zeros_like(prototype)
 
 
+class _Real(_SingletonConstraint):
+    def __call__(self, x):
+        # XXX: consider to relax this condition to [-inf, inf] interval
+        return (x == x) & (x != float("inf")) & (x != float("-inf")) & np.isreal(x)
```

Comment:
Could you remove `np.isreal` here? We don't check for real input across all constraints. If x is a tracer, np.isreal might not work.